bundles/org.openhab.binding.gardena/src/main/java/org/openhab/binding/gardena/internal/handler/GardenaThingHandler.java

```diff
@@ -261,17 +261,18 @@ private GardenaCommand getGardenaCommand(String dataItemProperty, ChannelUID cha
         String commandName = channelUID.getIdWithoutGroup().toUpperCase();
         String groupId = channelUID.getGroupId();
         if (groupId != null) {
-            if (groupId.startsWith("valve") && groupId.endsWith("_commands")) {
-                return new ValveCommand(ValveControl.valueOf(commandName),
-                        getDevice().getLocalService(dataItemProperty).commandDuration);
-            } else if ("mower_commands".equals(groupId)) {
-                return new MowerCommand(MowerControl.valueOf(commandName),
-                        getDevice().getLocalService(dataItemProperty).commandDuration);
-            } else if ("valveSet_commands".equals(groupId)) {
-                return new ValveSetCommand(ValveSetControl.valueOf(commandName));
-            } else if ("powerSocket_commands".equals(groupId)) {
-                return new PowerSocketCommand(PowerSocketControl.valueOf(commandName),
-                        getDevice().getLocalService(dataItemProperty).commandDuration);
+            switch (groupId) {
+                case "valve_commands":
```

Comment:
I don't know this code beforehand, so I think you'll need to explain the fix to me. 🙂 In the linked issue you wrote:
> the type should be VALE_SET_CONTROL in the json model

This line was the only actual change I was able to find (refactoring aside), but I don't get it.

I suppose the previous code somehow was taking this into consideration:
https://github.com/openhab/openhab-addons/blob/297640aa774556cfbf70044321ff90bd3b21e8d1/bundles/org.openhab.binding.gardena/src/main/resources/OH-INF/thing/thing-types.xml#L132-L164

I.e. "valveOne_commands" would be recognized and a `ValveCommand` would be returned.

I will also let @andrewfg and @gerrieg have a look.

Reply:
Yep, you are right. This condition was also true to the groupid = valveSet_commands, causing the valveSetCommand block to never be called.
```if (groupId.startsWith("valve") && groupId.endsWith("_commands")) {```

Looking at it again, i must have mixed up my production/development systems while testing, because the groupid=valveOne_commands fails. 

I just made a fix, re-order the if statement, and (with a little shame) i have re-tested it and now it is ok for both commands.